### PR TITLE
firstboot: Check if the given shell exists

### PR DESCRIPTION
**Pull request by daandemeyer**
cc @poettering 

Not entirely sure if this usage of `prefix_roota` is safe.

### CODE REVIEW

File: src/firstboot/firstboot.c [hunk 1]
```diff
@@ -604,6 +604,22 @@ static int prompt_root_password(void) {
         return 0;
 }
 
+static int find_shell(const char *path, const char *root)
+{
```
Comment (keszybz) [inline src/firstboot/firstboot.c]: Move "{" to previous line please.

File: src/firstboot/firstboot.c [hunk 2]
```diff
@@ -604,6 +604,22 @@ static int prompt_root_password(void) {
...
+        int r;
+
+        assert(path);
+
+        if (!valid_shell(path))
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "%s is not a valid shell", path);
+
+        r = chase_symlinks(path, root, CHASE_PREFIX_ROOT, NULL, NULL);
+        if (r < 0)
+                return log_error_errno(r, "Failed to resolve shell %s: %m", prefix_roota(root, path));
```
Comment (keszybz) [inline src/firstboot/firstboot.c]: alloca cannot be called in arguments to a function. So a temp variable needs to be declared, and alloca called, and then the variable can be used in the function call argument.

**Comment by keszybz** [comment]
LGTM.

**Comment by poettering** [comment]
lgtm, too. thanks for doing this!

**Pull request merged**